Packs/Threatgraph/Integrations/Threatgraph/Threatgraph.py

```diff
@@ -0,0 +1,234 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+"""
+    Crowdstrike have deprecated the ThreatGraph API but no replacement is ready
+    This integration replicates the get-tree REST call which provides
+    parent/child sensor/process information including src/dest IP and type information
+
+    Version History:
+    # v2.1.4 add back in conversion to strings for ids after changing to argToList and fix lint warnings
+    # v2.1.3 lint tidy
+    # v2.1.2 no change, just reversioned following git restructuring and changed tags for deployment
+    # v1.1.1 fix bugs where multiple items not working
+    # v1.1.0 Threatgraph api for the deprecated Crowdstrike Threatgraph
+    # as soon as Crowdstrike create a replacement then this integration can be removed
+"""
+
+import requests
+
+BASE_URL = demisto.params()['url'][:-1] if demisto.params()['url'].endswith('/') else demisto.params()['url']
+USERNAME = demisto.params().get('apikey')
+PASSWORD = demisto.params().get('apisecret')
+
+
+def get_tree(args):
+    """ the main entry point, for a given list of sensors and process ids
+        find the graph of resources associated with these and return
+
+    Args:
+        args (_type_): array of sensor_id and process_id, as defined in the integration
+
+    Returns:
+        _type_: dictionary of parent/child process information along with IP and type details
+    """
+    sensor_ids = argToList(args.get("sensor_ids"))
+    process_ids = argToList(args.get("process_ids"))
+
+    all_results: Dict[str, Any] = {}
+    all_ips = []
+    all_sources = []
+    all_destinations = []
+    all_key_data = ""
+
+    for sensor_id in sensor_ids:
+        sensor_id_str = str(sensor_id)
+        for process_id in process_ids:
+            process_id_str = str(process_id)
```

Comment:
Hello, from our experience they are not. If you do not use str() then you get cast errors, for example in concatenating strings to URLs - which we've already experienced in your Resilient integration. In addition, in an unrelated Palo bug we raised regarding List parameters you can see numerics are passed in because xsoar automation returns string to int casting errors.

Reply:
If it is a number we use arg_to_number function.
But if the data comes from the context that may happen if not used correctly. If this is the case you are trying to handle, please add a comment as future developer might find it redundant and remove.